app/assets/javascripts/app/datepicker-data.coffee

```diff
@@ -1,9 +1,15 @@
+class window.DatePickerData
+  @activate: ->
+    # This will set up a datepicker based on data attributes
+    # Make sure you call `to_s` or `iso8601` on any dates you set in the views
+    # to ensure the proper format.
+    $picker = $(".datepicker__data")
+
+    $picker.datepicker(
+      minDate: new Date($picker.data("min-date")) # will be unbounded if not provided
+      maxDate: new Date($picker.data("max-date")) # will be unbounded if not provided
+    )
+
 $ ->
-  # This will set up a datepicker based on data attributes
-  # Make sure you call `to_s` or `iso8601` on any dates you set in the views to
-  # ensure the proper format.
-  $picker = $(".datepicker__data")
-  $picker.datepicker(
-    minDate: new Date($picker.data("min-date")) # will be unbounded if not provided
-    maxDate: new Date($picker.data("max-date")) # will be unbounded if not provided
-  )
+  DatePickerData.activate()
```

Comment:
We're invoking it once here. The next line is just a reference to the function. That reference will get called whenever a modal gets loaded (after the ajax has finished) so that the datepicker gets bound to the new elements that were loaded on the page.